src/test/java/dev/drugowick/threehundredsixty/controller/FeedbackControllerIntegrationTest.java

```diff
@@ -0,0 +1,45 @@
+package dev.drugowick.threehundredsixty.controller;
+
+import dev.drugowick.threehundredsixty.service.MyUserDetailsService;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+class FeedbackControllerIntegrationTest {
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private MyUserDetailsService myUserDetailsService;
+
+    Long nonexistentFeedbackId = 9990009999L;
+    Long existentFeedbackId = 1L;
+
+    @Test
+    @WithMockUser(username = "enrique.iglesias@email.com", password = "password")
+    void whenGetNonExistentFeedback_thenFail() throws Exception {
+        mockMvc.perform(get("/feedback/" + nonexistentFeedbackId))
+                .andDo(print())
+                .andExpect(status().is5xxServerError());
```

Comment:
Eu tô curtindo esses esquemas de testes, mas não manjo o suficiente pra te criticar. O que posso colaborar é o seguinte: tem coisas aqui que estão meio que te atrapalhando, e esse é um exemplo. 

Esperar erro 500 não é legal, hahahaa. E só dá erro 500 por que foi mal implementado, por ser PoC, claro. O certo seria tratar esse negócio e retornar um erro decente. Aqui tem um excelente exemplo: https://algafoodapi.herokuapp.com/cities/10

Note que a cidade 10 não existe mas a API (nesse caso é uma API) retorna um 404 claramente explicando que a cidade não existe.

Minha sugestão é fazer os testes nesses moldes. Quem sabe até a gente devesse fazer num esquema TDD. Desenvolve o que esperamos da aplicação e depois a gente ajusta o código-fonte de acordo.

Reply:
Sim @brunodrugowick , concordo 100% com seu comentário sobre o http-500.
O lance é que eu escolhi não misturar criação de testes, com refactoring (pelo menos sem estarmos de comum acordo :joy:)

A ideia do TDD eu gostei! Não sei muito bem como por em prática aqui... 
¿deixaria os testes quebrados (e mergidos) até refatorar ou faria num mesmo commit? **Como você pensou?** 


----
> Sobre Além disso, fiquei com a impressão de que lançar uma RuntimeException genérica (feedbacks não encontrados) é pior para testar do que usar @ControllerAdvice para capturar elas... pois é... o ideal seria mesmo ter um @ControllerAdvice tratando de acordo com as exceções que fazem sentido do ponto de vista do negócio.
> 
> Aqui tem um excelente exemplo: https://github.com/brunodrugowick/algafood-api/blob/master/src/main/java/dev/drugowick/algaworks/algafoodapi/api/exceptionhandler/ApiExceptionHandler.java

Ótimo exemplo! Eu não sei bem o que esse _algafood_ faz, mas tá ficando o bixão haha
Eu tinha visto esse pattern para gerenciar exceptions [num post do medium](https://medium.com/@jovannypcg/understanding-springs-controlleradvice-cd96a364033f
), legal ver que você também aplicou.

Reply:
@filiagees o algafood é de um curso muito bom da Algaworks, foco em REST APIs. O código é mais deles, mas tem uma ou outra coisa que eu fiz por conta.

Sobre o TDD, acho que podemos fazer num outro repo, mesmo, pois a ideia já era refazer todo o código dessa PoC.

Mas se for pra fazer aqui, por mim poderia fazer e deixar os testes falharem, mesmo. OU